Move Helm chart helper scripts to scripts/ subdir

eero-t opened the pull request:
## Description

Move current `helm-charts/*.sh` scripts to new `helm-charts/scripts/` subdir.  Remove variable name inconsistency for that dir.

Done in preparation for future PRs that add more Helm chart helper scripts.

## Issues

`n/a`.

## Type of change

- [x] Enhancement

## Dependencies

`n/a`.

## Tests

Manually verified that all paths to these scripts have been updated accordingly:
```
scripts=$(ls helm-charts/scripts/)
for i in $scripts; do echo $i:; git grep $i; echo; done
```

## Next steps

Couple of docs in other repos reference `update_dependency.sh`, but that's all.  Those can be updated after this PR is in.

GenAIExamples:
```
DocSum/kubernetes/helm/README.md:./update_dependency.sh
DocSum/kubernetes/helm/README.md:./update_dependency.sh
```
docs:
```
tutorial/ChatQnA/deploy/k8s_getting_started.md:- A script called **./update_dependency.sh** is provided which is used to update chart dependencies, ensuring all nested charts are at their latest versions.
tutorial/ChatQnA/deploy/k8s_helm.md:./update_dependency.sh
```

poussa reviewed (approved):
LGTM. CI/CD needs to be fixed,though.

mkbhanda commented:
> There's something broken in CI path/link checker, it complains about 3 working (and already existing) links in Helm charts main `README.md`, and 1 working link in `TDX.md`.
> 
> Btw. according to `git grep check_values.sh`, that script is not referenced in any other file in any of the OPEA repositories. What's its purpose?

The script itself has not changed in over 7 months .. https://github.com/opea-project/GenAIInfra/blame/main/.github/workflows/pr-path-detection.yml

@ZePan110 would you kindly take a look.

eero-t commented:
> The script itself has not changed in over 7 months .. https://github.com/opea-project/GenAIInfra/blame/main/.githubworkflows/pr-path-detection.yml I was not any the wiser despite re-running the test with debug on.

I extracted this to separate script, and added output for the error codes.

The problematic README links get on first try 429 (Too Many Requests), and on retry 301 (Moved Permanently).

Replies seem to be coming from Fortinet-Proxy, i.e. its configuration is problem for the CI.

eero-t commented:
Another problem with the CI script are the reported document paths:
```
**********Validation failed, try again**********
Invalid link from /home/runner/work/GenAIInfra/GenAIInfra/README.md: https://github.com/opea-project/GenAIExamples/tree/main/CodeTrans/README.md
**********Validation failed, try again**********
Invalid link from /home/runner/work/GenAIInfra/GenAIInfra/README.md: https://github.com/opea-project/GenAIExamples/tree/main/DocSum/README.md
**********Validation failed, try again**********
Invalid link from /home/runner/work/GenAIInfra/GenAIInfra/README.md: https://github.com/opea-project/GenAIExamples/tree/main/VisualQnA/README.md
**********Validation failed, try again**********
Invalid link from /home/runner/work/GenAIInfra/GenAIInfra/TDX.md: https://github.com/canonical/tdx/blob/noble-24.04/README.md#setup-host-os
```

They are wrong (missing path within the repo):
```
GenAIInfra$ git grep -l -e CodeTrans/README.md -e DocSum/README.md -e VisualQnA/README.md -e setup-host-os
helm-charts/README.md
helm-charts/TDX.md
```

eero-t commented:
For some reason hyperlink checker is not run anymore (although I added commit changing it), so CI passes now.

PS. Why [Check Online Document Building](https://github.com/opea-project/GenAIInfra/actions/runs/14838451336/job/41655207077?pr=1041) test builds an image, instead of just testing building of the documentation?

eero-t commented:
Ah, CI passed because I added bug to the link checker script (YAML had a TAB in it).  That feels so wrong... :-)

eero-t commented:
I've locally tried adding 1-2s sleep between the URL checks, and alternative between normal and HEAD requests, but proxy still gives 429 (too many request) errors on those (valid) URLs.

eero-t commented:
After some iteration, found Fortinet proxy to block at least some URL request unless there's at least 15s delay between them (14s was _not_ enough) => test passes now.

 Fixed also the issue that if doc included identical link multiple times, script checked it multiple times.

Do you want these CI test script changes in a separate PR?

yongfengdu commented:
We'll need to sync values.yaml files from GenAIInfra to GenAIExamples/Comps for helm-chart deployment, this script is to help check any difference between these values.
> Btw. according to `git grep check_values.sh`, that script is not referenced in any other file in any of the OPEA repositories. What's its purpose?

yongfengdu commented:
> After some iteration, found Fortinet proxy to block at least some URL request unless there's at least 15s delay between them (14s was _not_ enough) => test passes now.
> 
> Fixed also the issue that if doc included identical link multiple times, script checked it multiple times.
> 
> Do you want these CI test script changes in a separate PR?
@ZePan110 Could you help review the pr-path-detection.yaml changes?
We can do this in one PR unless more changes are required for the pr-path part.

ZePan110 commented:
> @ZePan110 Could you help review the pr-path-detection.yaml changes? We can do this in one PR unless more changes are required for the pr-path part.
> 
> > After some iteration, found Fortinet proxy to block at least some URL request unless there's at least 15s delay between them (14s was _not_ enough) => test passes now.
> > Fixed also the issue that if doc included identical link multiple times, script checked it multiple times.
> > Do you want these CI test script changes in a separate PR?

ZePan110 commented:
Sorry for clicking close incorrectly 
LGTM
